packages/core/elasticsearch/core-elasticsearch-server-internal/src/version_check/ensure_es_version.ts

```diff
@@ -149,10 +150,25 @@ export const pollEsNodesVersion = ({
   log,
   kibanaVersion,
   ignoreVersionMismatch,
-  esVersionCheckInterval: healthCheckInterval,
+  healthCheckInterval,
+  healthCheckStartupInterval,
 }: PollEsNodesVersionOptions): Observable<NodesVersionCompatibility> => {
   log.debug('Checking Elasticsearch version');
-  return timer(0, healthCheckInterval).pipe(
+
+  const hasStartupInterval =
+    healthCheckStartupInterval !== undefined && healthCheckStartupInterval !== healthCheckInterval;
+
+  const isStartup$ = new BehaviorSubject(hasStartupInterval);
+
+  const checkInterval$ = isStartup$.pipe(
+    distinctUntilChanged(),
+    map((useStartupInterval) =>
+      useStartupInterval ? healthCheckStartupInterval! : healthCheckInterval
+    )
+  );
+
+  return checkInterval$.pipe(
+    switchMap((checkInterval) => timer(0, checkInterval)),
```

Comment:
The actual interval swap, always fun using `rxjs`

Reply:
When switching to `useStartupInterval: false`, there will be an emission immediately after the last one.

To workaround this, we could use `interval` + `startWith` instead of the `timer`. Something like:
<img width="686" alt="image" src="https://github.com/elastic/kibana/assets/25349407/4e6d1b22-5d02-4cb2-a027-1fcdc5fbbb6b">
WDYT?